Platform/RecursiveLock.swift

```diff
@@ -29,6 +29,24 @@ import class Foundation.NSRecursiveLock
             _ = Resources.decrementTotal()
         }
     }
+
+    /// A recursive lock for static instances. It doesn't increase or decrease the resource count
+    /// on initialization and deallocation.
+    ///
+    /// Use this class for static lock instances or the resource count will never decrease since
```

Comment:
Hey :) 
I'm not sure `the resource count will never decrease` is the correct statement here. Because it would decrease on lock/unlock it would just never decrease the one incremented on `init` right? That's at least what I could understand, maybe I'm getting this wrong :)
Other thing is if the instance is always allocated, isn't correct to have a resource count always taking it into account? I mean have it incremented would be correct?

Reply:
Hey, thanks for your comment. 

This behavior might be theoretically correct. But in perspective of that [`Resources.total` API is designed for debugging memory leaks](https://github.com/ReactiveX/RxSwift/blob/6b2a406b928cc7970874dcaed0ab18e7265e41ef/Documentation/GettingStarted.md#debugging-memory-leaks), this might be confused because the resource count remains on `1` even if every resources **_seems to be_** deallocated.

Reply:
That makes sense,  the comment `never decremented` got me a little bit confused.
Thank's for the answer :))

Reply:
I added some words to make more sense!

```diff
- will never decrease
+ will never decrease under 1
```